lib/backend/Memory.js

```diff
@@ -239,14 +294,14 @@ export default class Memory {
             const minScore = (min === '-inf') ? this.data[key][0][0] : min;
             const maxScore = (min === '+inf') ?
                 this.data[key][this.data[key].length][0] : max;
-            let removeCount = 0;
-            this.data[key].forEach((item, index) => {
+            const oldLen = this.data[key].length;
+            this.data[key] = this.data[key].filter(item => {
                 if (item[0] >= minScore && item[0] <= maxScore) {
-                    this.data[key].splice(index, 1);
-                    removeCount++;
+                    return false;
```

Comment:
Since you only have a conditional in that filter, no need to keep the returning boolean clutter, you can just use lambda syntax.